File: second_task/app/src/main/java/com/example/secondtask/main/second_fragment/ShowFragment.java
package com.example.secondtask.main.second_fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import com.example.secondtask.R;
import com.example.secondtask.main.model.TetaMenza;
import com.example.secondtask.main.view_model.CustomViewModel;

import java.util.List;

public class ShowFragment extends Fragment {

    private ListView listView;
    private CustomViewModel viewModel;
    private ArrayAdapter<TetaMenza> adapter;
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(this.requireActivity()).get(CustomViewModel.class);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.show_fragment,container,false);

        listView = rootView.findViewById(R.id.list_view);
        adapter = new ArrayAdapter<>(getContext(), android.R.layout.simple_list_item_1);
        listView.setAdapter(adapter);
        viewModel.getTete().observe(this.getActivity(), new Observer<List<TetaMenza>>() {
            @Override
            public void onChanged(List<TetaMenza> strings) {
                adapter.addAll(strings);
                adapter.notifyDataSetChanged();

            }
        });

        return rootView;
    }
}

File: second_task/app/src/main/java/com/example/secondtask/main/second_framgment/ShowFragment.kt
package com.example.secondtask.main.second_framgment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.lifecycle.Observer
import com.example.secondtask.R
import com.example.secondtask.databinding.InputFragmentBinding
import com.example.secondtask.databinding.ShowFragmentBinding
import com.example.secondtask.main.model.TetaMenza
import com.example.secondtask.main.view_model.CustomViewModel

class ShowFragment : Fragment() {

    private val viewModel : CustomViewModel by activityViewModels()
    private var _binding : ShowFragmentBinding? = null
    private val binding get() = _binding!!
    private lateinit var adapter : ArrayAdapter<TetaMenza>

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = ShowFragmentBinding.inflate(inflater,container,false)
        val view = binding.root
        adapter= ArrayAdapter(context!!, R.layout.custom_list_item)
        binding.listView.adapter = adapter
        viewModel.tete.observe(viewLifecycleOwner, Observer {
            adapter.addAll(it)
            adapter.notifyDataSetChanged()
        })
        return view
    }
}